### ISSUE TITLE
False Positives dartanalyzer:prefer_relative_imports with package Imports

### ISSUE THREAD

**Issue by virtualmarc** [insideapp-oss/sonar-flutter#18]
We are getting many false positives in SonarQube for the rule `dartanalyzer:prefer_relative_imports`.

The rule marks every import of  third party libary imports, e.g.:

```dart
import 'package:path/path.dart';
```

Running dartanalyzer manually we get "No issues found", also the IDE is not marking those imports as an Error.

**Comment by stephanecodes** [comment]
If you have a `analysis_options.yaml` under the project root where `prefer_relative_imports` is unset or set to false, then `dartanalyzer` will use it and won't report any issue.

But the plugin enforce the use of [its own analysis options](https://github.com/insideapp-oss/sonar-flutter/blob/master/dart-lang/src/main/resources/fr/insideapp/sonarqube/dart/dartanalyzer/analysis_options.yaml) file where `prefer_relative_imports` rule is enabled.

That's why it's reported in SonarQube.

**Comment by virtualmarc** [comment]
My `analysis_options.yaml` has `prefer_relative_imports` enabled.
The IDE and the dartanalyzer CLI also mark non-relative imports for files in my own project. (Everything correct here).

But SonarQube marks all external packages with this rule. 
So, if I have an import to, e.g. sqflite. The IDE and CLI do not show any error, but sonarqube marks it with an error and tells me to make it a relative import, which is not possible, since it's an external dependency.

I had to deactivate the rule in SonarQube, because I got loads of false-positives.

**Comment by zippy1978** [comment]
Hi @virtualmarc,

As @stephanecodes said, the plugin uses its own options during the analysis, to make sure all issues are captured.
If you don't want them to show up in your reports : what you did (disabling it in the SonarQube profile) is the right thing to do.

**Comment by zippy1978** [comment]
Anyway, I merged PR https://github.com/insideapp-oss/sonar-flutter/pull/20 by @stephanecodes.

You can now add 
`sonar.dart.analysis.useExistingOptions=true`
in your sonar-project.properties to keep yours.

**Comment by zippy1978** [comment]
I released version 0.3.0 with **sonar.dart.analysis.useExistingOptions** support.
I close the issue.
